blocks/algorithms/__init__.py

```diff
@@ -710,29 +710,32 @@ class RemoveNotFinite(StepRule):
 
     Replaces a step (the parameter update of a single shared variable)
     which contains non-finite elements (such as ``inf`` or ``NaN``) with a
-    scaled version of the parameters being updated instead.
+    step rescaling the parameters.
 
     Parameters
     ----------
     scaler : float, optional
         The scaling applied to the parameter in case the step contains
-        non-finite elements. Defaults to 0.1.
+        non-finite elements. Defaults to 1, which means that parameters
+        will not be changed.
 
     Notes
     -----
+    This rule should be applied last!
+
     This trick was originally used in the GroundHog_ framework.
 
     .. _GroundHog: https://github.com/lisa-groundhog/GroundHog
 
     """
-    def __init__(self, scaler=0.1):
+    def __init__(self, scaler=1):
         self.scaler = scaler
 
     def compute_step(self, param, previous_step):
         not_finite = tensor.any(tensor.or_(
```

Comment:
Should probably file a bug on Theano for that one.

On Fri, May 22, 2015 at 1:08 PM, Orhan Firat <notifications@github.com>
wrote:

> In blocks/algorithms/__init__.py
> <https://github.com/mila-udem/blocks/pull/658#discussion_r30912800>:
>
> >      This trick was originally used in the GroundHog_ framework.
> >
> >      .. _GroundHog: https://github.com/lisa-groundhog/GroundHog
> >
> >      """
> > -    def __init__(self, scaler=0.1):
> > +    def __init__(self, scaler=1):
> >          self.scaler = scaler
> >
> >      def compute_step(self, param, previous_step):
> >          not_finite = tensor.any(tensor.or_(
>
> We realized with @kyunghyuncho <https://github.com/kyunghyuncho> that
> tensor.any here is slowing down the overall iteration with a significant
> amount. For instance in a large computation graph running on gpu, by using
> BLOCKS_PROFILE=True we were getting
> with tensor.any 13.5 secs per iteration
> without tensor.any 1.9 secs per iteration
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mila-udem/blocks/pull/658/files#r30912800>.
>